src/parser/index.js

```diff
@@ -273,17 +273,25 @@ class Parser {
    * @return {Node}
    */
   primary () {
-    if (this.matches('NUMBER', 'STRING')) {
-      let type = this.previous().type === 'NUMBER'
-        ? 'NumberLiteral'
-        : 'StringLiteral'
-
+    if (this.matches('NUMBER')) {
       return {
-        type,
+        type: 'NumberLiteral',
         value: this.previous().value
       }
     }
 
+    if (this.matches('QUOTE')) {
```

Comment:
Do we need an error in case the paren isn't closed? Might be missing something

Reply:
`matches` will consume the ending `QUOTE` as well, or error out if it reaches EOF before that

Reply:
Or well, I see it doesn't actually error out, it goes into an infinite loop. I will add a `strict` parameter to `matches` to have it error out if it reaches EOF before finding a match.